python/lsst/ts/hexgui/application.py

```diff
@@ -0,0 +1,162 @@
+    """
+
+    # The set of "aboutToQuit" comes from "qasync/examples/aiohttp_fetch.py"
+    loop = asyncio.get_event_loop()
+    future: asyncio.Future = asyncio.Future()
+
+    # You need one (and only one) QApplication instance per application.
+    # There is one QApplication instance in "qasync" already.
+    app = qasync.QApplication.instance()
+    if hasattr(app, "aboutToQuit"):
+        getattr(app, "aboutToQuit").connect(
+            functools.partial(close_future, future, loop)
+        )
+
+    app.setApplicationName("Hexapod EUI")
+
+    # Set the parser
+    parser = QCommandLineParser()
+    parser.setApplicationDescription("Run the hexapod graphical user interface (GUI).")
+    parser.addHelpOption()
+
+    parser.addPositionalArgument("hexapod", "1: camera hexapod, 2: M2 hexapod.")
+
+    option_verbose = QCommandLineOption(
+        ["v", "verbose"], "Print log messages to terminal."
+    )
+    parser.addOption(option_verbose)
+
+    option_simulation = QCommandLineOption(
+        ["s", "simulation"], "Run the simulation mode."
+    )
+    parser.addOption(option_simulation)
+
+    option_log_level = QCommandLineOption(
+        ["d", "debuglevel"],
+        (
+            "Debug logging level: CRITICAL (50), ERROR (40), WARNING (30), "
+            "INFO (20), DEBUG (10), NOTSET (0). The default is 20."
+        ),
+        "level",
+        "20",
+    )
+    parser.addOption(option_log_level)
+
+    option_no_log_file = QCommandLineOption(
+        ["no-logfile"], "Do not write log messages to file."
+    )
+    parser.addOption(option_no_log_file)
+
+    parser.process(app)
+
+    # Get the argument and check the values
+    args = parser.positionalArguments()
+    check_arguments(args)
+
+    is_output_log_to_file = not parser.isSet(option_no_log_file)
+    is_verbose = parser.isSet(option_verbose)
+    is_simulation_mode = parser.isSet(option_simulation)
+    log_level = int(parser.value(option_log_level))
+
+    # Create a Qt main window, which will be our window.
+    window_main = MainWindow(
+        is_output_log_to_file,
+        is_verbose,
+        MTHexapod.SalIndex(int(args[0])),
+        is_simulation_mode,
+        log_level=log_level,
+    )
+    window_main.show()
+
+    await future
+
+    # Your application won't reach here until you exit and the event
+    # loop has stopped.
+    return True
+
+
+def check_arguments(args: list) -> None:
+    """Check the arguments.
+
+    Parameters
+    ----------
+    args : `list`
+        Arguments from the command line.
+
+    Raises
+    ------
+    `ValueError`
+        If the number of arguments is not 1.
+    `ValueError`
+        If the argument is not 1 or 2.
+    """
+
+    if len(args) != 1:
+        raise ValueError("The number of arguments must be 1.")
+
+    if args[0] not in ["1", "2"]:
+        raise ValueError("The argument must be 1 or 2.")
+
+
+def close_future(future: asyncio.Future, loop: asyncio.AbstractEventLoop) -> None:
```

Comment:
The function is called `close_future` but you actually cancel the future. That is confusing. Please consider renaming this function.

Reply:
Updated in the commit: [36c6b48](https://github.com/lsst-ts/ts_hexgui/pull/1/commits/36c6b4888cbec3cdec2c5ceff4b4ac4bd97a3ee2). Thanks!